core/src/main/java/org/apache/iceberg/util/Tasks.java

```diff
@@ -88,8 +89,17 @@ public UnrecoverableException(Throwable cause) {
     private long maxSleepTimeMs = 600000; // 10 minutes
     private long maxDurationMs = 600000;  // 10 minutes
     private double scaleFactor = 2.0;     // exponential
+    private boolean hasStarted = false;
 
+    /**
+     * @deprecated Use {@link #Builder(Iterator)} instead.
```

Comment:
> I think you can use `@deprecated since 0.14.0`.

I don't think it adds value. If a user is < 0.14, they won't see deprecation notice.
If they are on >= 0.14, they see it. 

It would be valuable if it comes with the "will be removed", but it's not my call to make promises like this.

Please advise what should i do.

Reply:
It doesn’t have to be added if it truly doesn’t add values.

We have some places where it’s not present, but I also tend to think of it as a notice for people who fork the project in case they’re using it.

I also think of it as something that reminds us (or specifically release managers) when something can be removed entirely.

Burr I agree here it wouldn’t present much end user facing value.